.pre-commit-config.yaml

```diff
@@ -0,0 +1,55 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+      - id: check-executables-have-shebangs
```

Comment:
Do we have something in CONTRIBUTING.md that describes the proper shebang
(`#!/usr/bin/env python3`)?

Reply:
No. I thought as executable files are made by us for mostly testing purposes, to include this in case we forget to add it.
Should I remove it?

Reply:
How many files will this flag?

Reply:
3 files:
- machine_learning/scoring_functions.py
- maths/sum_of_arithmetic_series.py
- scheduling/round_robin.py

Reply:
Let’s leave it in.

Reply:
So, shall I include the shebang in those files or remove the executable bit from them?